src/mip/utils/serial_port.c

```diff
@@ -249,14 +250,20 @@ bool serial_port_read(serial_port *port, void *buffer, size_t num_bytes, int wai
     if(!serial_port_is_open(port))
         return false;
 
+    uint32_t bytes_available = serial_port_read_count(port);
+
 #ifdef WIN32 //Windows
 
     if( wait_time <= 0 )
     {
-        if( serial_port_read_count(port) == 0 )
+        if(bytes_available == 0 )
             return true;
     }
 
+    //Don't let windows block on the read
+    if(bytes_available < num_bytes)
+        num_bytes = bytes_available;
+
```

Comment:
It should still block on bytes_available == 0 though. This allows the thread to sleep while waiting for data. How about `if(bytes_available > 0 && bytes_available < num_bytes)`?